server/src/main/java/org/cloudfoundry/identity/uaa/scim/vmidentity/VmidentityScimUserProvisioning.java

```diff
@@ -104,115 +109,59 @@ public ScimUser delete(String id, int version) {
 
-        if ( scimFilter.getFilterType() == SCIMFilterType.OR )
-        {
-            scimFilters = scimFilter.getFilterComponents();
-        }
-        else if ( scimFilter.getFilterType() == SCIMFilterType.EQUALITY )
-        {
-            scimFilters = new ArrayList<SCIMFilter>();
-            scimFilters.add(scimFilter);
-        }
-        else
-        {
-            logger.debug(String.format("Scim filter '%s' not yet supported", filter));
-            throw new UnsupportedOperationException(String.format("Scim filter '%s' not yet supported", filter));
+            return scimUsers;
+        } catch (Exception e) {
+            logger.error("Unable to query users by filter '" + filter + "'", e);
+            throw new VmidentityDataAccessException("Unable to query users by filter: " + filter);
         }
+    }
 
-        for(SCIMFilter f : scimFilters)
-        {
-            // for now only support equality by id or username
-            if ( f.getFilterType() == SCIMFilterType.EQUALITY )
-            {
-                if ( ( "id".equalsIgnoreCase(f.getFilterAttribute().getAttributeName()) )
-                    || "username".equalsIgnoreCase(f.getFilterAttribute().getAttributeName()))
-                {
-                    if ( f.getFilterValue() != null )
-                    {
-                        users.add(this.retrieve(f.getFilterValue()));
-                    }
-                }
-                else
-                {
-                    logger.debug(String.format("Scim filter '%s' attribute '%s' not yet supported",
-                            filter, f.getFilterAttribute().getAttributeName()));
-                    throw new UnsupportedOperationException(
-                        String.format("Scim filter '%s' attribute '%s' not yet supported",
-                            filter, f.getFilterAttribute().getAttributeName()));
+    @Override
+    public List<ScimUser> query(String filter, String sortBy, boolean ascending) {
+        List<ScimUser> users = query(filter);
 
-                }
-            }
-            else
-            {
-                logger.debug(String.format("Scim filter '%s' not yet supported", filter));
-                throw new UnsupportedOperationException(String.format("Scim filter '%s' not yet supported" , filter));
-            }
+        if (sortBy != null) {
+            sortScimUsers(users, sortBy, ascending);
         }
         return users;
     }
 
     @Override
-    public List<ScimUser> query(String filter, String sortBy, boolean ascending)
-    {
-        // todo: sort by, ascending
-        return this.query(filter);
-    }
-
-    @Override
     public int delete(String filter) {
-        throw new UnsupportedOperationException();
+        logger.debug("Filtering users with query: " + filter);
+
+        try {
+            String tenant = VmidentityUtils.getTenantName(idmClient);
+            Set<PersonUser> users = idmClient.findPersonUsersByScimFilter(tenant, filter);
```

Comment:
I think it should probably be person users only. I need to look elsewhere to see where we might be also using solution users.